aodndata/aatams/aatams_sattag.py

```diff
@@ -0,0 +1,171 @@
+        )
+
+    def process_nrt(self):
+        """Process NRT files, only allowing updates to occur if the new file
+        is more recent."""
+        previous_files = self.state_query.query_storage(self.dest_path_function(""))
+        if not previous_files:
+            return
+
+        old_metadata_file = self.get_metadata_file(previous_files)
+        new_metadata_file = self.get_metadata_file(self.file_collection)
+        logger.info(
+            NRT_TIMESTAMP_COMPARISON_MSG
+            % (old_metadata_file.dest_path, new_metadata_file.src_path)
+        )
+        #TODO: remove this compatibility layer when aodncore is updated
+        try:
+            self.state_query.download(
+                RemotePipelineFileCollection(old_metadata_file), self.temp_dir
+            )
+        except AttributeError:
+            self.state_query._storage_broker.download(
+                RemotePipelineFileCollection(old_metadata_file), (self.temp_dir)
+            )
+
+        if not self.is_nrt_update_required(old_metadata_file.local_path):
+            raise InvalidFileContentError(
+                NRT_TIMESTAMP_DIFFERS_MSG
+                % (
+                    new_metadata_file.src_path,
+                    self.input_file_object.src_path,
+                    old_metadata_file.dest_path,
+                )
+            )
+        # TODO: Enable below to schedule files to be deleted, if required
+        # for remote_file in previous_files:
+        #     self.schedule_file_removal(remote_file)
+
+    def preprocess(self):
+        """Validate the received file(s) and set publish types"""
+        if self.input_file_object.extension == ".csv":
+            self.schema.validate_file(self.input_file_object.src_path)
+            self.input_file_object.publish_type = PipelineFilePublishType.HARVEST_UPLOAD
+        else:
+            files_in_zip = self.file_collection.get_attribute_list("local_path")
+            self.validation_call(files_in_zip)
+
+            self.input_file_object.publish_type = PipelineFilePublishType.ARCHIVE_ONLY
+
+            self.file_collection.filter_by_attribute_value(
+                "file_type", FileType.CSV,
+            ).set_publish_types(PipelineFilePublishType.HARVEST_UPLOAD)
+
+    def process(self):
+        """ Process NRT files if this class is initialize as NRT pipeline """
+        if self.nrt_aware():
+            self.process_nrt()
+
+
+# AATAMS DM global constants
+AATAMS_SATTAG_QC_DM_BASE = "IMOS/AATAMS/satellite_tagging/ATF_Location_QC_DM"
+AATAMS_SATTAG_QC_DM_OPTS = {
+    "allowed_extensions": [".zip", ".csv"],
+    "allowed_regexes": ["^.+_dm\\.(zip|csv)$"],
+    "allowed_archive_path_regexes": [AATAMS_SATTAG_QC_DM_BASE + ".+\\.(zip|csv)"],
+    "allowed_dest_path_regexes": [AATAMS_SATTAG_QC_DM_BASE + ".+\\.(zip|csv)"],
+    "archive_input_file": True,
+    "dest_path_function": aatams_sattag_qc_dm_dest_path,
+    "archive_path_function": aatams_sattag_qc_dm_dest_path,
```

Comment:
@evacougnon @ocehugo Just to agree on this, if you push the files to archive with the same filename, they will be overwritten. I imagine we're missing the point of keeping different versions of them. IMO, we should add a timestamp of when they got uploaded, and maybe the uploader name if possible (my point is in case we repush a file ourselves), but I don't think we can retrieve this info, so probably doesn't matter

Anyway, S3 is an object storage so ... We usually push files to archive if we create a product from the incoming file, but in this case we simply push them, so I don't really see the advantage. @bpasquer Do you have an opinion on the matter since you're pretty much the archive expert !

Reply:
I don't understand, their locations are different, so they shall not be overwritten.